specification/resources/DeploymentStacks.Management/models-whatIf.tsp

```diff
@@ -0,0 +1,379 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "./models.tsp";
+
+using Azure.ResourceManager;
+using Azure.ResourceManager.Foundations;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+using Microsoft.Resources.DeploymentStacks;
+
+namespace Microsoft.Resources.DeploymentStacks;
+
+/** Deployment stack object. */
+@added(Versions.v2025_07_01)
+model DeploymentStacksWhatIfResult
+  is Azure.ResourceManager.ProxyResource<DeploymentStacksWhatIfResultProperties> {
+  ...DeploymentStacksWhatIfResultNameParameter;
+
+  /** The geo-location where the resource lives. Required for subscription and management group scoped stacks. The location is inherited from the resource group for resource group scoped stacks. */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "Model is shared between resource group, subscription, and management group scope."
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  location?: string;
+
+  /** Resource tags. */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "This is the expected type"
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "Model is shared between resource group, subscription, and management group scope."
+  tags?: Record<string>;
+}
+
+alias DeploymentStacksWhatIfResultNameParameter = ResourceNameParameter<
+  Resource = DeploymentStacksWhatIfResult,
+  KeyName = "deploymentStacksWhatIfResultName",
+  SegmentName = "deploymentStacksWhatIfResults",
+  NamePattern = "^[-\\w\\._\\(\\)]+$"
+>;
+
+@@maxLength(DeploymentStacksWhatIfResult.name, 90);
+@@minLength(DeploymentStacksWhatIfResult.name, 1);
+@@doc(DeploymentStacksWhatIfResult.name,
+  "Name of the deployment stack what-if result."
+);
+
+/**
+ * DeploymentStack WhatIfResult Properties
+ */
+@added(Versions.v2025_07_01)
+model DeploymentStacksWhatIfResultProperties is DeploymentStacksError {
+  ...DeploymentStackSharedProperties;
+
+  /**
+   * The deployment stack id to use as the basis for comparison.
+   */
+  deploymentStackResourceId: string;
```

Comment:
Are we going to allow this to be modified after the what-if resource is created? If not, do we need to set a lifecycle annotation to indicate that it's immutable after creation?